src/main/java/uk/gov/hmcts/ccd/domain/types/DocumentValidator.java

```diff
@@ -74,24 +96,128 @@ public BaseType getType() {
 
         if (dataValue.has(DOCUMENT_BINARY_URL)) {
             final JsonNode documentBinaryUrl = dataValue.get(DOCUMENT_BINARY_URL);
-
-            if (!documentBinaryUrl.isTextual() || documentBinaryUrl.isNull()) {
-                return Collections.singletonList(new ValidationResult(
-                    "document_binary_url is not a text value or is null", dataFieldId));
+            validationResults = validateDocumentBinaryURL(
+                dataFieldId,documentBinaryUrl,urlPattern,documentUrlValue, urlPatternString);
+            if (!validationResults.isEmpty()) {
+                return validationResults;
             }
+        }
 
-            final String documentBinaryUrlValue = documentBinaryUrl.textValue();
-            final Matcher documentBinaryUrlMatcher = urlPattern.matcher(documentBinaryUrlValue);
+        if (dataValue.has(CATEGORY_ID)) {
+            validationResults = validateCategoryId(dataFieldId,dataValue,caseFieldDefinition);
+            if (!validationResults.isEmpty()) {
+                return validationResults;
+            }
+        }
 
-            if (!documentBinaryUrlMatcher.matches()) {
-                LOG.error("{} does not match Document Management domain or expected URL path {} {}",
-                          documentUrlValue, urlPatternString, dataFieldId);
-                return Collections.singletonList(new ValidationResult(documentBinaryUrlValue
-                    + " does not match Document Management domain or expected URL path", dataFieldId));
+        if (dataValue.has(UPLOAD_TIMESTAMP)) {
+            final JsonNode uploadTimeStamp = dataValue.get(UPLOAD_TIMESTAMP);
+            validationResults = validateUploadTimeStamp(dataFieldId,uploadTimeStamp,caseFieldDefinition);
+            if (!validationResults.isEmpty()) {
+                return validationResults;
             }
         }
 
         return Collections.emptyList();
     }
 
+    private List<ValidationResult> validateDocumentBinaryURL(final String dataFieldId,
+                                                             final JsonNode documentBinaryUrl,
+                                                             Pattern urlPattern,
+                                                             String documentUrlValue,
+                                                             String urlPatternString) {
+        isNullOrEmptyResult = isNullOrEmpty(documentBinaryUrl,DOCUMENT_BINARY_URL,dataFieldId);
+        if (!isNullOrEmptyResult.isEmpty()) {
+            return isNullOrEmptyResult;
+        }
```

Comment:
For the scenarios in this class where we're checking null or empty it might be a bit cleaner/easier to understand if doing something like this instead:
```suggestion
        if (isNullOrEmpty(documentBinaryUrl)) {
            return nullOrEmptyValidationResult(DOCUMENT_BINARY_URL, dataFieldId);
        }
```
where that new method returns the common validation result
```
private List<ValidationResult> nullOrEmptyValidationResult(String key, String dataFieldId) {
        LOG.debug("{} is not a text value or is null", key);
        return Collections.singletonList(new ValidationResult(
            key + " is not a text value or is empty", dataFieldId));
}
```